NOTICK gradle and script work

blsemo opened the pull request:
- Move gradle files to groovy from kotlin DSL
- Add maven publish so we can publish the API at least to local
- Add version 0.0.1-beta
- Add shell script to start CLI from POSIX systems
- Improve Windows start script
- Upgrade log4j to 2.17

blsemo committed:
- Move gradle files to groovy from kotlin DSL
- Add maven publish so we can publish the API at least to local
- Add version 0.0.1-beta
- Add shell script to start CLI from POSIX systems
- Improve Windows start script
- Upgrade log4j to 2.17

chrisr3 reviewed (commented):
The Kotlin script was crude, but the Groovy version does not need to be.

chrisr3 committed:
Upgrade to Gradle 7.3.2, configure the Kotlin compiler and remove cruft.

chrisr3 committed:
Publish API jar as corda-cli-api.

chrisr3 committed:
Make implicit Kotlin library dependency explicit, and use picoCliVersion property.

chrisr3 committed:
Declare repositories using Gradle 7.3's dependencyResolutionManagement.

chrisr3 reviewed (changes requested):
These logging dependencies make _no sense_.

blsemo committed:
Review comments

chrisr3 reviewed (commented):
This project is making my eyes bleed...

blsemo committed:
More review fixes (but didn't find a solution around the
jar.get() call - however this does not have an impact on
gradle task performance).

blsemo committed:
Add multi release to make log4j work correctly with stackwalker

blsemo committed:
Clean up syntax

chrisr3 reviewed (approved):
LGTM. Well, good enough :wink:.

blsemo committed:
Move mgm plug-in to groovy

blsemo committed:
Switch back to just put jars in lib
